test/e2e/framework/ingress_utils.go

```diff
@@ -865,13 +865,19 @@ func (cont *GCEIngressController) GetFirewallRuleName() string {
 
 // GetFirewallRule returns the firewall used by the GCEIngressController.
 func (cont *GCEIngressController) GetFirewallRule() *compute.Firewall {
-	gceCloud := cont.Cloud.Provider.(*gcecloud.GCECloud)
-	fwName := cont.GetFirewallRuleName()
-	fw, err := gceCloud.GetFirewall(fwName)
+	fw, err := cont.GetFirewallRuleOrError()
 	Expect(err).NotTo(HaveOccurred())
 	return fw
 }
 
+// GetFirewallRule returns the firewall used by the GCEIngressController.
+// Returns an error if that fails.
+func (cont *GCEIngressController) GetFirewallRuleOrError() (*compute.Firewall, error) {
+	gceCloud := cont.Cloud.Provider.(*gcecloud.GCECloud)
```

Comment:
Remove OrError

Reply:
There is already an existing GetFirewallRule method that asserts that the error should be nil. This one returns an error.

Reply:
I agree with @bowei. See my comment above.